### PR TITLE
created checkout and visitors list routes

### PR DESCRIPTION
**Pull request by abibou1**
Checkout route:
checkout

Visistors:
visitors

### CODE REVIEW

File: server/src/api/v1/routes/visitors.js [hunk 1]
```diff
@@ -87,4 +87,45 @@ router.post('/admin/visitors/checkin', authenticatedAdmin, async (req, res) => {
...
+    "hostID": "",
+    "tag": "",
+    "visitorID": "",
+    "timeIn": "",
+    "timeOut": ""
+}
+*/
+router.post('/admin/visitors/checkout', authenticatedAdmin, async (req, res) => {
+  try {
+    const resp = await Visit.updateOne({ tag: req.body.tag }, { $set: { timeOut: Date.now() } });
```
Comment (mazma1) [inline server/src/api/v1/routes/visitors.js:106]: Before you perform this update, I think it is best to confirm that this tag belongs to the visitor who wants to be check out.
- payload should include email and tag
- find the visit you want to update using the tag
- find the full visitor details using the visitorID from the visit record you want to update
- if the email from the visitor details matches the email you were given, then perform the update
- if they do not match, return an error saying there is no record of that tag for the provided email

File: server/src/api/v1/routes/visitors.js [hunk 2]
```diff
@@ -87,4 +87,45 @@ router.post('/admin/visitors/checkin', authenticatedAdmin, async (req, res) => {
...
+    // response.n === 0, then no update was made.
+    if (resp.n === 0) res.status(403).json({ error: 'The tag is invalid.' });
+
+    const updateVisit = await Visit.findOne({ tag: req.body.tag });
```
Comment (mazma1) [inline server/src/api/v1/routes/visitors.js:110]: Just a side note, there was no need making this request again. `resp` and `updateVisit` will basically be the same thing

**Review by mazma1 — COMMENTED** [review]
The logic for checkout needs to be tightened up a bit, let me know if my suggestion makes sense

File: server/src/api/v1/routes/visitors.js (hunk 1 above)
Comment (abibou1) [reply server/src/api/v1/routes/visitors.js]: This is a good idea. Thank you!

File: server/src/api/v1/routes/visitors.js (hunk 2 above)
Comment (abibou1) [reply server/src/api/v1/routes/visitors.js:118]: To my surprise, UpdateOne's response is not the object modified. The response is like below:
updateONeResponse

I console.log by replacing the line #116 on visitors.js with:
```
const resp = await Visit.updateOne({ tag: req.body.tag }, { $set: { timeOut: Date.now() } });
console.log(`Resp${JSON.stringify(resp)}`);
```

Here is the final result:
checkout1

Comment (mazma1) [reply server/src/api/v1/routes/visitors.js:118]: Interesting, this is new. Thanks for clarifying the different responses.

File: server/src/api/v1/routes/visitors.js [hunk 3]
```diff
@@ -87,4 +87,53 @@ router.post('/admin/visitors/checkin', authenticatedAdmin, async (req, res) => {
...
+
+    const findVisitor = await Visitor.findOne({ _id: findVisit.visitorID });
+    if (findVisitor.email !== email) {
+      res.status(403).json({ error: 'No record for this combined tag and email.' });
+    }
+
+    await Visit.updateOne({ tag: req.body.tag }, { $set: { timeOut: Date.now() } });
+
+    const updateVisit = await Visit.findOne({ tag: req.body.tag });
+    res.status(202).json({
```
Comment (mazma1) [inline server/src/api/v1/routes/visitors.js:119]: I'm curious why this is a `202`, which basically means that this request has been accepted but not completed

Comment (abibou1) [reply server/src/api/v1/routes/visitors.js]: Sorry, I didn't know that 202 mean that the request is not completed. Next time, I will read all the details before using a new HTTP status.

**Pull request merged**